In [1]:
print('radhe radhe')

radhe radhe


In [2]:
!pip install -q google-generativeai

In [3]:
pip install datasets

Defaulting to user installation because normal site-packages is not writeable
   ---------------------------------------- 0.0/510.5 kB ? eta -:--:--
   -- ------------------------------------ 30.7/510.5 kB 640.0 kB/s eta 0:00:01
   -- ------------------------------------ 30.7/510.5 kB 640.0 kB/s eta 0:00:01
   --- ----------------------------------- 41.0/510.5 kB 326.8 kB/s eta 0:00:02
   ---- ---------------------------------- 61.4/510.5 kB 363.1 kB/s eta 0:00:02
   ---- ---------------------------------- 61.4/510.5 kB 363.1 kB/s eta 0:00:02
   ------- ------------------------------- 92.2/510.5 kB 403.5 kB/s eta 0:00:02
   ------- ------------------------------- 92.2/510.5 kB 403.5 kB/s eta 0:00:02
   ------------ ------------------------- 163.8/510.5 kB 517.2 kB/s eta 0:00:01
   ------------ ------------------------- 163.8/510.5 kB 517.2 kB/s eta 0:00:01
   ------------ ------------------------- 163.8/510.5 kB 517.2 kB/s eta 0:00:01
   ------------ ------------------------- 163.8/510

In [5]:
import datasets
gsm8k = datasets.load_dataset('gsm8k','main')


Generating train split:   0%|          | 0/7473 [00:00<?, ? examples/s]

Generating test split:   0%|          | 0/1319 [00:00<?, ? examples/s]

In [23]:
print(len(gsm8k['test']))

1319


In [24]:
import pandas as pd
# Convert the dataset to a pandas DataFrame
df = pd.DataFrame(gsm8k['test'])  

# Export the DataFrame to a CSV file
df.to_csv('gsm8k_test1.csv', index=False)  

print("CSV file exported successfully.")

CSV file exported successfully.


In test set we have total 1319 tets examples of question as well as their detailed answer.But due to the limitation of compuatation power we will randomly sample only 200 examples for our model evaluation

In [25]:
import pandas as pd
import random

# Read the original CSV file into a pandas DataFrame
df = pd.read_csv('gsm8k_test1.csv')

# Randomly select 200 rows from the DataFrame
random_indices = random.sample(range(len(df)), 200)
df_selected = df.iloc[random_indices]

# Write the selected rows to a new CSV file named 'gsm8k_test.csv'
df_selected.to_csv('gsm8k_test.csv', index=False)

print("Randomly selected 200 rows written to 'gsm8k_test.csv' successfully.")

Randomly selected 200 rows written to 'gsm8k_test.csv' successfully.


In [32]:
df=pd.read_csv('gsm8k_test.csv')
print(df.shape)
df.head()

(200, 2)


question  \
0  Ali has four $10 bills and six $20 bills that ...   
1  Shiela bought five cell phones for $150 each f...   
2  Rita hand-picks Junebugs off of her plants eve...   
3  While at the dollar store, Sloane counts 100 c...   
4  Three friends: Mike, Jim, and Tony decided to ...   

                                              answer  
0  Four $10 bills have a value of 4*$10 =$<<4*10=...  
1  A $150 x 2/100 = $<<150*2/100=3>>3 interest wi...  
2  On both Tuesday and Wednesday, she removed twi...  
3  On the second day, she counted 100+50 = <<100+...  
4  After 3 rounds Jim has 21 points - 3 points = ...

In [28]:
import google.generativeai as palm

In [29]:
palm.configure(api_key="AIzaSyCV6bQxB4C4HD2AjaBfyKJOiEX8ZTw8Ggg")

In [30]:
models=[m for m in palm.list_models() if "generateText" in m.supported_generation_methods]

for m in models:
    print(f"Model Name: {m.name}")

Model Name: models/text-bison-001


In [31]:
model=models[0].name
model

'models/text-bison-001'

 # Let's start to experiment with our GSM8K data set

## 8-Shot Example with CoT prompt

In [22]:
import pandas as pd



# Open the text file in write mode
with open('evaluation_PaLM.txt', 'w',encoding="utf-8") as file:
    # Iterate over each row in the DataFrame
    for index, row in df.iterrows():
        question = row['question']
        answer = row['answer']
        prompt=f'''
        After looking at the following 8 questions and their detailed answers  , write a step-by-step explanation on how to solve the 9th question:
        Question1: "There are 15 trees in the grove. Grove workers will plant trees in the grove today. After they are done, there will be 21 trees. How many trees did the grove workers plant today?"
        Answer: "There are 15 trees originally. Then there were 21 trees after some more were planted. So there must have been 21 - 15 = 6. The answer is 6."
        Question2: "If there are 3 cars in the parking lot and 2 more cars arrive, how many cars are in the parking lot?"
        Answer: "There are originally 3 cars. 2 more cars arrive. 3 + 2 = 5. The answer is 5."
        Question3: "Leah had 32 chocolates and her sister had 42. If they ate 35, how many pieces do they have left in total?"
        Answer: "Originally, Leah had 32 chocolates. Her sister had 42. So in total they had 32 + 42 = 74. After eating 35, they had 74 - 35 = 39. The answer is 39."
        Question4: "Jason had 20 lollipops. He gave Denny some lollipops. Now Jason has 12 lollipops. How many lollipops did Jason give to Denny?"
        Answer: "Jason started with 20 lollipops. Then he had 12 after giving some to Denny. So he gave Denny 20 - 12 = 8. The answer is 8."
        Question5: "Shawn has five toys. For Christmas, he got two toys each from his mom and dad. How many toys does he have now?"
        Answer: "Shawn started with 5 toys. If he got 2 toys each from his mom and dad, then that is 4 more toys. 5 + 4 = 9.The answer is 9."
        Question6: "There were nine computers in the server room. Five more computers were installed each day, from monday to thursday. How many computers are now in the server room?"
        Answer: "There were originally 9 computers. For each of 4 days, 5 more computers were added. So 5 * 4 = 20 computers were added. 9 + 20 is 29. The answer is 29."
        Question7: "Michael had 58 golf balls. On tuesday, he lost 23 golf balls. On wednesday, he lost 2 more. How many golf balls did he have at the end of wednesday?"
        Answer: "Michael started with 58 golf balls. After losing 23 on tuesday, he had 58 - 23 = 35. After losing 2 more, he had 35 - 2 = 33 golf balls. The answer is 33."
        Question8: "Olivia has $23. She bought five bagels for $3 each. How much money does she have left?"
        Answer: "Olivia had 23 dollars. 5 bagels for 3 dollars each will be 5 x 3 = 15 dollars. So she has 23 - 15 dollars left. 23- 15 is 8. The answer is 8."
        Question9: {question}
        '''
        completion=palm.generate_text(model=model,prompt=prompt,temperature=0.5,max_output_tokens=1024)
        model_answer=completion.result
        # Write the question and the new answer to the text file
        file.write(f"QUESTION: {question}\n***********************************\nANSWER: {answer}\n**********************************\nMODEL's_ANSWER: {model_answer}\n##############################################################################################\n")
        
       
        
   


print("Text file 'evaluation_PaLM.txt' created successfully.")


Text file 'evaluation_PaLM.txt' created successfully.


In [33]:
import pandas as pd



# Open the text file in write mode
with open('evaluation_PaLM_8shot_CoT.txt', 'w',encoding="utf-8") as file:
    # Iterate over each row in the DataFrame
    for index, row in df.iterrows():
        question = row['question']
        answer = row['answer']
        prompt=f'''
        After looking at the following 8 questions and their detailed answers  , write a solution with step-by-step explanation for the 9th question:
        Question1: "There are 15 trees in the grove. Grove workers will plant trees in the grove today. After they are done, there will be 21 trees. How many trees did the grove workers plant today?"
        Answer: "There are 15 trees originally. Then there were 21 trees after some more were planted. So there must have been 21 - 15 = 6. The answer is 6."
        Question2: "If there are 3 cars in the parking lot and 2 more cars arrive, how many cars are in the parking lot?"
        Answer: "There are originally 3 cars. 2 more cars arrive. 3 + 2 = 5. The answer is 5."
        Question3: "Leah had 32 chocolates and her sister had 42. If they ate 35, how many pieces do they have left in total?"
        Answer: "Originally, Leah had 32 chocolates. Her sister had 42. So in total they had 32 + 42 = 74. After eating 35, they had 74 - 35 = 39. The answer is 39."
        Question4: "Jason had 20 lollipops. He gave Denny some lollipops. Now Jason has 12 lollipops. How many lollipops did Jason give to Denny?"
        Answer: "Jason started with 20 lollipops. Then he had 12 after giving some to Denny. So he gave Denny 20 - 12 = 8. The answer is 8."
        Question5: "Shawn has five toys. For Christmas, he got two toys each from his mom and dad. How many toys does he have now?"
        Answer: "Shawn started with 5 toys. If he got 2 toys each from his mom and dad, then that is 4 more toys. 5 + 4 = 9.The answer is 9."
        Question6: "There were nine computers in the server room. Five more computers were installed each day, from monday to thursday. How many computers are now in the server room?"
        Answer: "There were originally 9 computers. For each of 4 days, 5 more computers were added. So 5 * 4 = 20 computers were added. 9 + 20 is 29. The answer is 29."
        Question7: "Michael had 58 golf balls. On tuesday, he lost 23 golf balls. On wednesday, he lost 2 more. How many golf balls did he have at the end of wednesday?"
        Answer: "Michael started with 58 golf balls. After losing 23 on tuesday, he had 58 - 23 = 35. After losing 2 more, he had 35 - 2 = 33 golf balls. The answer is 33."
        Question8: "Olivia has $23. She bought five bagels for $3 each. How much money does she have left?"
        Answer: "Olivia had 23 dollars. 5 bagels for 3 dollars each will be 5 x 3 = 15 dollars. So she has 23 - 15 dollars left. 23- 15 is 8. The answer is 8."
        Question9: {question}
        '''
        completion=palm.generate_text(model=model,prompt=prompt,temperature=0.5,max_output_tokens=1024)
        model_answer=completion.result
        # Write the question and the new answer to the text file
        file.write(f"QUESTION: {question}\n***********************************\nANSWER: {answer}\n**********************************\nMODEL's_ANSWER: {model_answer}\n##############################################################################################\n")
        
       
        
   


print("Text file 'evaluation_PaLM_8shot_CoT.txt' created successfully.")

Text file 'evaluation_PaLM_8shot_CoT.txt' created successfully.


## 7 Shot Example of CoT prompt

In [35]:
# Open the text file in write mode
with open('evaluation_PaLM_7shot_CoT.txt', 'w',encoding="utf-8") as file:
    # Iterate over each row in the DataFrame
    for index, row in df.iterrows():
        question = row['question']
        answer = row['answer']
        prompt=f'''
        After looking at the following 7 questions and their detailed answers, write a solution with step-by-step explanation for the 8th question:
        Question1: "There are 15 trees in the grove. Grove workers will plant trees in the grove today. After they are done, there will be 21 trees. How many trees did the grove workers plant today?"
        Answer: "There are 15 trees originally. Then there were 21 trees after some more were planted. So there must have been 21 - 15 = 6. The answer is 6."
        Question2: "If there are 3 cars in the parking lot and 2 more cars arrive, how many cars are in the parking lot?"
        Answer: "There are originally 3 cars. 2 more cars arrive. 3 + 2 = 5. The answer is 5."
        Question3: "Leah had 32 chocolates and her sister had 42. If they ate 35, how many pieces do they have left in total?"
        Answer: "Originally, Leah had 32 chocolates. Her sister had 42. So in total they had 32 + 42 = 74. After eating 35, they had 74 - 35 = 39. The answer is 39."
        Question4: "Jason had 20 lollipops. He gave Denny some lollipops. Now Jason has 12 lollipops. How many lollipops did Jason give to Denny?"
        Answer: "Jason started with 20 lollipops. Then he had 12 after giving some to Denny. So he gave Denny 20 - 12 = 8. The answer is 8."
        Question5: "Shawn has five toys. For Christmas, he got two toys each from his mom and dad. How many toys does he have now?"
        Answer: "Shawn started with 5 toys. If he got 2 toys each from his mom and dad, then that is 4 more toys. 5 + 4 = 9.The answer is 9."
        Question6: "There were nine computers in the server room. Five more computers were installed each day, from monday to thursday. How many computers are now in the server room?"
        Answer: "There were originally 9 computers. For each of 4 days, 5 more computers were added. So 5 * 4 = 20 computers were added. 9 + 20 is 29. The answer is 29."
        Question7: "Michael had 58 golf balls. On tuesday, he lost 23 golf balls. On wednesday, he lost 2 more. How many golf balls did he have at the end of wednesday?"
        Answer: "Michael started with 58 golf balls. After losing 23 on tuesday, he had 58 - 23 = 35. After losing 2 more, he had 35 - 2 = 33 golf balls. The answer is 33."

        Question8: {question}
        '''
        completion=palm.generate_text(model=model,prompt=prompt,temperature=0.5,max_output_tokens=1024)
        model_answer=completion.result
        # Write the question and the new answer to the text file
        file.write(f"QUESTION: {question}\n***********************************\nANSWER: {answer}\n**********************************\nMODEL's_ANSWER: {model_answer}\n##############################################################################################\n")
        
       
        
   


print("evaluation_PaLM_7shot_CoT.txt' created successfully.")

InternalServerError: 500 An internal error has occurred. Please retry or report in https://developers.generativeai.google/guide/troubleshooting

## 6 Shot CoT Example

In [ ]:
# Open the text file in write mode
with open('evaluation_PaLM_6shot_CoT.txt', 'w',encoding="utf-8") as file:
    # Iterate over each row in the DataFrame
    for index, row in df.iterrows():
        question = row['question']
        answer = row['answer']
        prompt=f'''
        After looking at the following 6 questions and their detailed answers  , write a solution with step-by-step explanation for the 7th question:
        Question1: "There are 15 trees in the grove. Grove workers will plant trees in the grove today. After they are done, there will be 21 trees. How many trees did the grove workers plant today?"
        Answer: "There are 15 trees originally. Then there were 21 trees after some more were planted. So there must have been 21 - 15 = 6. The answer is 6."
        Question2: "If there are 3 cars in the parking lot and 2 more cars arrive, how many cars are in the parking lot?"
        Answer: "There are originally 3 cars. 2 more cars arrive. 3 + 2 = 5. The answer is 5."
        Question3: "Leah had 32 chocolates and her sister had 42. If they ate 35, how many pieces do they have left in total?"
        Answer: "Originally, Leah had 32 chocolates. Her sister had 42. So in total they had 32 + 42 = 74. After eating 35, they had 74 - 35 = 39. The answer is 39."
        Question4: "Jason had 20 lollipops. He gave Denny some lollipops. Now Jason has 12 lollipops. How many lollipops did Jason give to Denny?"
        Answer: "Jason started with 20 lollipops. Then he had 12 after giving some to Denny. So he gave Denny 20 - 12 = 8. The answer is 8."
        Question5: "Shawn has five toys. For Christmas, he got two toys each from his mom and dad. How many toys does he have now?"
        Answer: "Shawn started with 5 toys. If he got 2 toys each from his mom and dad, then that is 4 more toys. 5 + 4 = 9.The answer is 9."
        Question6: "There were nine computers in the server room. Five more computers were installed each day, from monday to thursday. How many computers are now in the server room?"
        Answer: "There were originally 9 computers. For each of 4 days, 5 more computers were added. So 5 * 4 = 20 computers were added. 9 + 20 is 29. The answer is 29."

        Question7: {question}
        '''
        completion=palm.generate_text(model=model,prompt=prompt,temperature=0.5,max_output_tokens=1024)
        model_answer=completion.result
        # Write the question and the new answer to the text file
        file.write(f"QUESTION: {question}\n***********************************\nANSWER: {answer}\n**********************************\nMODEL's_ANSWER: {model_answer}\n##############################################################################################\n")
        
       
        
   


print("evaluation_PaLM_6shot_CoT.txt' created successfully.")

## 5 shot CoT example

In [ ]:
# Open the text file in write mode
with open('evaluation_PaLM_5shot_CoT.txt', 'w',encoding="utf-8") as file:
    # Iterate over each row in the DataFrame
    for index, row in df.iterrows():
        question = row['question']
        answer = row['answer']
        prompt=f'''
        After looking at the following 5 questions and their detailed answers  , write a solution with step-by-step explanation for the 6th question:
        Question1: "There are 15 trees in the grove. Grove workers will plant trees in the grove today. After they are done, there will be 21 trees. How many trees did the grove workers plant today?"
        Answer: "There are 15 trees originally. Then there were 21 trees after some more were planted. So there must have been 21 - 15 = 6. The answer is 6."
        Question2: "If there are 3 cars in the parking lot and 2 more cars arrive, how many cars are in the parking lot?"
        Answer: "There are originally 3 cars. 2 more cars arrive. 3 + 2 = 5. The answer is 5."
        Question3: "Leah had 32 chocolates and her sister had 42. If they ate 35, how many pieces do they have left in total?"
        Answer: "Originally, Leah had 32 chocolates. Her sister had 42. So in total they had 32 + 42 = 74. After eating 35, they had 74 - 35 = 39. The answer is 39."
        Question4: "Jason had 20 lollipops. He gave Denny some lollipops. Now Jason has 12 lollipops. How many lollipops did Jason give to Denny?"
        Answer: "Jason started with 20 lollipops. Then he had 12 after giving some to Denny. So he gave Denny 20 - 12 = 8. The answer is 8."
        Question5: "Shawn has five toys. For Christmas, he got two toys each from his mom and dad. How many toys does he have now?"
        Answer: "Shawn started with 5 toys. If he got 2 toys each from his mom and dad, then that is 4 more toys. 5 + 4 = 9.The answer is 9."
        Question6: {question}
        '''
        completion=palm.generate_text(model=model,prompt=prompt,temperature=0.5,max_output_tokens=1024)
        model_answer=completion.result
        # Write the question and the new answer to the text file
        file.write(f"QUESTION: {question}\n***********************************\nANSWER: {answer}\n**********************************\nMODEL's_ANSWER: {model_answer}\n##############################################################################################\n")
        
       
        
   


print("Text file 'evaluation_PaLM_5shot_CoT.txt' created successfully.")

## 4 shot CoT Example

In [ ]:
# Open the text file in write mode
with open('evaluation_PaLM_4shot_CoT.txt', 'w',encoding="utf-8") as file:
    # Iterate over each row in the DataFrame
    for index, row in df.iterrows():
        question = row['question']
        answer = row['answer']
        prompt=f'''
        After looking at the following 4 questions and their detailed answers  , write a solution with step-by-step explanation for the 5th question:
        Question1: "There are 15 trees in the grove. Grove workers will plant trees in the grove today. After they are done, there will be 21 trees. How many trees did the grove workers plant today?"
        Answer: "There are 15 trees originally. Then there were 21 trees after some more were planted. So there must have been 21 - 15 = 6. The answer is 6."
        Question2: "If there are 3 cars in the parking lot and 2 more cars arrive, how many cars are in the parking lot?"
        Answer: "There are originally 3 cars. 2 more cars arrive. 3 + 2 = 5. The answer is 5."
        Question3: "Leah had 32 chocolates and her sister had 42. If they ate 35, how many pieces do they have left in total?"
        Answer: "Originally, Leah had 32 chocolates. Her sister had 42. So in total they had 32 + 42 = 74. After eating 35, they had 74 - 35 = 39. The answer is 39."
        Question4: "Jason had 20 lollipops. He gave Denny some lollipops. Now Jason has 12 lollipops. How many lollipops did Jason give to Denny?"
        Answer: "Jason started with 20 lollipops. Then he had 12 after giving some to Denny. So he gave Denny 20 - 12 = 8. The answer is 8."

        Question5: {question}
        '''
        completion=palm.generate_text(model=model,prompt=prompt,temperature=0.5,max_output_tokens=1024)
        model_answer=completion.result
        # Write the question and the new answer to the text file
        file.write(f"QUESTION: {question}\n***********************************\nANSWER: {answer}\n**********************************\nMODEL's_ANSWER: {model_answer}\n##############################################################################################\n")
        
       
        
   


print("Text file 'evaluation_PaLM_4shot_CoT.txt' created successfully.")

## 3 Shot CoT Example

In [ ]:
# Open the text file in write mode
with open('evaluation_PaLM_3shot_CoT.txt', 'w',encoding="utf-8") as file:
    # Iterate over each row in the DataFrame
    for index, row in df.iterrows():
        question = row['question']
        answer = row['answer']
        prompt=f'''
        After looking at the following 3 questions and their detailed answers  , write a solution with step-by-step explanation for the 4th question:
        Question1: "There are 15 trees in the grove. Grove workers will plant trees in the grove today. After they are done, there will be 21 trees. How many trees did the grove workers plant today?"
        Answer: "There are 15 trees originally. Then there were 21 trees after some more were planted. So there must have been 21 - 15 = 6. The answer is 6."
        Question2: "If there are 3 cars in the parking lot and 2 more cars arrive, how many cars are in the parking lot?"
        Answer: "There are originally 3 cars. 2 more cars arrive. 3 + 2 = 5. The answer is 5."
        Question3: "Leah had 32 chocolates and her sister had 42. If they ate 35, how many pieces do they have left in total?"
        Answer: "Originally, Leah had 32 chocolates. Her sister had 42. So in total they had 32 + 42 = 74. After eating 35, they had 74 - 35 = 39. The answer is 39."


        Question4: {question}
        '''
        completion=palm.generate_text(model=model,prompt=prompt,temperature=0.5,max_output_tokens=1024)
        model_answer=completion.result
        # Write the question and the new answer to the text file
        file.write(f"QUESTION: {question}\n***********************************\nANSWER: {answer}\n**********************************\nMODEL's_ANSWER: {model_answer}\n##############################################################################################\n")
        
       
        
   


print("Text file 'evaluation_PaLM_3shot_CoT.txt' created successfully.")

## 2 Shot CoT Example

In [ ]:
# Open the text file in write mode
with open('evaluation_PaLM_2shot_CoT.txt', 'w',encoding="utf-8") as file:
    # Iterate over each row in the DataFrame
    for index, row in df.iterrows():
        question = row['question']
        answer = row['answer']
        prompt=f'''
        After looking at the following 2 questions and their detailed answers  , write a solution with step-by-step explanation for the 3rd question:
        Question1: "There are 15 trees in the grove. Grove workers will plant trees in the grove today. After they are done, there will be 21 trees. How many trees did the grove workers plant today?"
        Answer: "There are 15 trees originally. Then there were 21 trees after some more were planted. So there must have been 21 - 15 = 6. The answer is 6."
        Question2: "If there are 3 cars in the parking lot and 2 more cars arrive, how many cars are in the parking lot?"
        Answer: "There are originally 3 cars. 2 more cars arrive. 3 + 2 = 5. The answer is 5."

        Question3: {question}
        '''
        completion=palm.generate_text(model=model,prompt=prompt,temperature=0.5,max_output_tokens=1024)
        model_answer=completion.result
        # Write the question and the new answer to the text file
        file.write(f"QUESTION: {question}\n***********************************\nANSWER: {answer}\n**********************************\nMODEL's_ANSWER: {model_answer}\n##############################################################################################\n")
        
       
        
   


print("Text file 'evaluation_PaLM_2shot_CoT.txt' created successfully.")

## 1 shot CoT Example

In [ ]:
# Open the text file in write mode
with open('evaluation_PaLM_1shot_CoT.txt', 'w',encoding="utf-8") as file:
    # Iterate over each row in the DataFrame
    for index, row in df.iterrows():
        question = row['question']
        answer = row['answer']
        prompt=f'''
        After looking at the following 1 question and its detailed answer  , write a solution with step-by-step explanation for the 2nd question:
        Question1: "There are 15 trees in the grove. Grove workers will plant trees in the grove today. After they are done, there will be 21 trees. How many trees did the grove workers plant today?"
        Answer: "There are 15 trees originally. Then there were 21 trees after some more were planted. So there must have been 21 - 15 = 6. The answer is 6."


        Question2: {question}
        '''
        completion=palm.generate_text(model=model,prompt=prompt,temperature=0.5,max_output_tokens=1024)
        model_answer=completion.result
        # Write the question and the new answer to the text file
        file.write(f"QUESTION: {question}\n***********************************\nANSWER: {answer}\n**********************************\nMODEL's_ANSWER: {model_answer}\n##############################################################################################\n")
        
       
        
   


print("Text file 'evaluation_PaLM_1shot_CoT.txt' created successfully.")

# O shot CoT Example

In [ ]:
# Open the text file in write mode
with open('evaluation_PaLM_0shot_CoT.txt', 'w',encoding="utf-8") as file:
    # Iterate over each row in the DataFrame
    for index, row in df.iterrows():
        question = row['question']
        answer = row['answer']
        prompt=f'''
        Provide a solution of the following question with step-by-step explanation:


        Question: {question}
        '''
        completion=palm.generate_text(model=model,prompt=prompt,temperature=0.5,max_output_tokens=1024)
        model_answer=completion.result
        # Write the question and the new answer to the text file
        file.write(f"QUESTION: {question}\n***********************************\nANSWER: {answer}\n**********************************\nMODEL's_ANSWER: {model_answer}\n##############################################################################################\n")
        
       
        
   


print("Text file 'evaluation_PaLM_0shot_CoT.txt' created successfully.")

## 8 shot standard prompt example

In [ ]:
import pandas as pd



# Open the text file in write mode
with open('evaluation_PaLM_8shot_standardPrompt.txt', 'w',encoding="utf-8") as file:
    # Iterate over each row in the DataFrame
    for index, row in df.iterrows():
        question = row['question']
        answer = row['answer']
        prompt=f'''
        After looking at the following 8 questions and their answers  , solve the 9th question:
        Question1: "There are 15 trees in the grove. Grove workers will plant trees in the grove today. After they are done, there will be 21 trees. How many trees did the grove workers plant today?"
        Answer: "The answer is 6."
        Question2: "If there are 3 cars in the parking lot and 2 more cars arrive, how many cars are in the parking lot?"
        Answer: "The answer is 5."
        Question3: "Leah had 32 chocolates and her sister had 42. If they ate 35, how many pieces do they have left in total?"
        Answer: "The answer is 39."
        Question4: "Jason had 20 lollipops. He gave Denny some lollipops. Now Jason has 12 lollipops. How many lollipops did Jason give to Denny?"
        Answer: "The answer is 8."
        Question5: "Shawn has five toys. For Christmas, he got two toys each from his mom and dad. How many toys does he have now?"
        Answer: "The answer is 9."
        Question6: "There were nine computers in the server room. Five more computers were installed each day, from monday to thursday. How many computers are now in the server room?"
        Answer: "The answer is 29."
        Question7: "Michael had 58 golf balls. On tuesday, he lost 23 golf balls. On wednesday, he lost 2 more. How many golf balls did he have at the end of wednesday?"
        Answer: "The answer is 33."
        Question8: "Olivia has $23. She bought five bagels for $3 each. How much money does she have left?"
        Answer: "The answer is 8."
        Question9: {question}
        '''
        completion=palm.generate_text(model=model,prompt=prompt,temperature=0.5,max_output_tokens=1024)
        model_answer=completion.result
        # Write the question and the new answer to the text file
        file.write(f"QUESTION: {question}\n***********************************\nANSWER: {answer}\n**********************************\nMODEL's_ANSWER: {model_answer}\n##############################################################################################\n")
        
       
        
   


print("Text file 'evaluation_PaLM.txt' created successfully.")

## 7 shot Standard Prompt Example

In [ ]:
import pandas as pd



# Open the text file in write mode
with open('evaluation_PaLM_7shot_standardPrompt.txt', 'w',encoding="utf-8") as file:
    # Iterate over each row in the DataFrame
    for index, row in df.iterrows():
        question = row['question']
        answer = row['answer']
        prompt=f'''
        After looking at the following 7 questions and their answers  , solve the 8th question:
        Question1: "There are 15 trees in the grove. Grove workers will plant trees in the grove today. After they are done, there will be 21 trees. How many trees did the grove workers plant today?"
        Answer: "The answer is 6."
        Question2: "If there are 3 cars in the parking lot and 2 more cars arrive, how many cars are in the parking lot?"
        Answer: "The answer is 5."
        Question3: "Leah had 32 chocolates and her sister had 42. If they ate 35, how many pieces do they have left in total?"
        Answer: "The answer is 39."
        Question4: "Jason had 20 lollipops. He gave Denny some lollipops. Now Jason has 12 lollipops. How many lollipops did Jason give to Denny?"
        Answer: "The answer is 8."
        Question5: "Shawn has five toys. For Christmas, he got two toys each from his mom and dad. How many toys does he have now?"
        Answer: "The answer is 9."
        Question6: "There were nine computers in the server room. Five more computers were installed each day, from monday to thursday. How many computers are now in the server room?"
        Answer: "The answer is 29."
        Question7: "Michael had 58 golf balls. On tuesday, he lost 23 golf balls. On wednesday, he lost 2 more. How many golf balls did he have at the end of wednesday?"
        Answer: "The answer is 33."
        Question8: {question}
        '''
        completion=palm.generate_text(model=model,prompt=prompt,temperature=0.5,max_output_tokens=1024)
        model_answer=completion.result
        # Write the question and the new answer to the text file
        file.write(f"QUESTION: {question}\n***********************************\nANSWER: {answer}\n**********************************\nMODEL's_ANSWER: {model_answer}\n##############################################################################################\n")
        
       
        
   


print("Text file 'evaluation_PaLM.txt' created successfully.")

## 6 shot Standard Prompt Example

In [ ]:
import pandas as pd



# Open the text file in write mode
with open('evaluation_PaLM_6shot_standardPrompt.txt', 'w',encoding="utf-8") as file:
    # Iterate over each row in the DataFrame
    for index, row in df.iterrows():
        question = row['question']
        answer = row['answer']
        prompt=f'''
        After looking at the following 6 questions and their answers  , solve the 7th question:
        Question1: "There are 15 trees in the grove. Grove workers will plant trees in the grove today. After they are done, there will be 21 trees. How many trees did the grove workers plant today?"
        Answer: "The answer is 6."
        Question2: "If there are 3 cars in the parking lot and 2 more cars arrive, how many cars are in the parking lot?"
        Answer: "The answer is 5."
        Question3: "Leah had 32 chocolates and her sister had 42. If they ate 35, how many pieces do they have left in total?"
        Answer: "The answer is 39."
        Question4: "Jason had 20 lollipops. He gave Denny some lollipops. Now Jason has 12 lollipops. How many lollipops did Jason give to Denny?"
        Answer: "The answer is 8."
        Question5: "Shawn has five toys. For Christmas, he got two toys each from his mom and dad. How many toys does he have now?"
        Answer: "The answer is 9."
        Question6: "There were nine computers in the server room. Five more computers were installed each day, from monday to thursday. How many computers are now in the server room?"
        Answer: "The answer is 29."

        Question7: {question}
        '''
        completion=palm.generate_text(model=model,prompt=prompt,temperature=0.5,max_output_tokens=1024)
        model_answer=completion.result
        # Write the question and the new answer to the text file
        file.write(f"QUESTION: {question}\n***********************************\nANSWER: {answer}\n**********************************\nMODEL's_ANSWER: {model_answer}\n##############################################################################################\n")
        
       
        
   


print("Text file 'evaluation_PaLM.txt' created successfully.")

## 5 shot standard prompt example

In [ ]:
import pandas as pd



# Open the text file in write mode
with open('evaluation_PaLM_5shot_standardPrompt.txt', 'w',encoding="utf-8") as file:
    # Iterate over each row in the DataFrame
    for index, row in df.iterrows():
        question = row['question']
        answer = row['answer']
        prompt=f'''
        After looking at the following 5 questions and their answers  , solve the 6th question:
        Question1: "There are 15 trees in the grove. Grove workers will plant trees in the grove today. After they are done, there will be 21 trees. How many trees did the grove workers plant today?"
        Answer: "The answer is 6."
        Question2: "If there are 3 cars in the parking lot and 2 more cars arrive, how many cars are in the parking lot?"
        Answer: "The answer is 5."
        Question3: "Leah had 32 chocolates and her sister had 42. If they ate 35, how many pieces do they have left in total?"
        Answer: "The answer is 39."
        Question4: "Jason had 20 lollipops. He gave Denny some lollipops. Now Jason has 12 lollipops. How many lollipops did Jason give to Denny?"
        Answer: "The answer is 8."
        Question5: "Shawn has five toys. For Christmas, he got two toys each from his mom and dad. How many toys does he have now?"
        Answer: "The answer is 9."

        Question6: {question}
        '''
        completion=palm.generate_text(model=model,prompt=prompt,temperature=0.5,max_output_tokens=1024)
        model_answer=completion.result
        # Write the question and the new answer to the text file
        file.write(f"QUESTION: {question}\n***********************************\nANSWER: {answer}\n**********************************\nMODEL's_ANSWER: {model_answer}\n##############################################################################################\n")
        
       
        
   


print("Text file 'evaluation_PaLM.txt' created successfully.")

## 4 shot standard prompt example

In [ ]:
import pandas as pd



# Open the text file in write mode
with open('evaluation_PaLM_4shot_standardPrompt.txt', 'w',encoding="utf-8") as file:
    # Iterate over each row in the DataFrame
    for index, row in df.iterrows():
        question = row['question']
        answer = row['answer']
        prompt=f'''
        After looking at the following 4 questions and their answers  , solve the 5th question:
        Question1: "There are 15 trees in the grove. Grove workers will plant trees in the grove today. After they are done, there will be 21 trees. How many trees did the grove workers plant today?"
        Answer: "The answer is 6."
        Question2: "If there are 3 cars in the parking lot and 2 more cars arrive, how many cars are in the parking lot?"
        Answer: "The answer is 5."
        Question3: "Leah had 32 chocolates and her sister had 42. If they ate 35, how many pieces do they have left in total?"
        Answer: "The answer is 39."
        Question4: "Jason had 20 lollipops. He gave Denny some lollipops. Now Jason has 12 lollipops. How many lollipops did Jason give to Denny?"
        Answer: "The answer is 8."

        Question5: {question}
        '''
        completion=palm.generate_text(model=model,prompt=prompt,temperature=0.5,max_output_tokens=1024)
        model_answer=completion.result
        # Write the question and the new answer to the text file
        file.write(f"QUESTION: {question}\n***********************************\nANSWER: {answer}\n**********************************\nMODEL's_ANSWER: {model_answer}\n##############################################################################################\n")
        
       
        
   


print("Text file 'evaluation_PaLM.txt' created successfully.")

## 3 shot standard prompt example

In [ ]:
import pandas as pd



# Open the text file in write mode
with open('evaluation_PaLM_3shot_standardPrompt.txt', 'w',encoding="utf-8") as file:
    # Iterate over each row in the DataFrame
    for index, row in df.iterrows():
        question = row['question']
        answer = row['answer']
        prompt=f'''
        After looking at the following 3 questions and their answers  , solve the 4th question:
        Question1: "There are 15 trees in the grove. Grove workers will plant trees in the grove today. After they are done, there will be 21 trees. How many trees did the grove workers plant today?"
        Answer: "The answer is 6."
        Question2: "If there are 3 cars in the parking lot and 2 more cars arrive, how many cars are in the parking lot?"
        Answer: "The answer is 5."
        Question3: "Leah had 32 chocolates and her sister had 42. If they ate 35, how many pieces do they have left in total?"
        Answer: "The answer is 39."

        Question4: {question}
        '''
        completion=palm.generate_text(model=model,prompt=prompt,temperature=0.5,max_output_tokens=1024)
        model_answer=completion.result
        # Write the question and the new answer to the text file
        file.write(f"QUESTION: {question}\n***********************************\nANSWER: {answer}\n**********************************\nMODEL's_ANSWER: {model_answer}\n##############################################################################################\n")
        
       
        
   


print("Text file 'evaluation_PaLM.txt' created successfully.")

## 2 shot standard prompt example

In [ ]:
import pandas as pd



# Open the text file in write mode
with open('evaluation_PaLM_2shot_standardPrompt.txt', 'w',encoding="utf-8") as file:
    # Iterate over each row in the DataFrame
    for index, row in df.iterrows():
        question = row['question']
        answer = row['answer']
        prompt=f'''
        After looking at the following 2 questions and their answers  , solve the 3rd question:
        Question1: "There are 15 trees in the grove. Grove workers will plant trees in the grove today. After they are done, there will be 21 trees. How many trees did the grove workers plant today?"
        Answer: "The answer is 6."
        Question2: "If there are 3 cars in the parking lot and 2 more cars arrive, how many cars are in the parking lot?"
        Answer: "The answer is 5."

        Question3: {question}
        '''
        completion=palm.generate_text(model=model,prompt=prompt,temperature=0.5,max_output_tokens=1024)
        model_answer=completion.result
        # Write the question and the new answer to the text file
        file.write(f"QUESTION: {question}\n***********************************\nANSWER: {answer}\n**********************************\nMODEL's_ANSWER: {model_answer}\n##############################################################################################\n")
        
       
        
   


print("Text file 'evaluation_PaLM_2shot_standardPrompt.txt' created successfully.")

## 1 shot standard prompt

In [ ]:
import pandas as pd



# Open the text file in write mode
with open('evaluation_PaLM_1shot_standardPrompt.txt', 'w',encoding="utf-8") as file:
    # Iterate over each row in the DataFrame
    for index, row in df.iterrows():
        question = row['question']
        answer = row['answer']
        prompt=f'''
        After looking at the following 1 question and its answer  , solve the 2nd question:
        Question1: "There are 15 trees in the grove. Grove workers will plant trees in the grove today. After they are done, there will be 21 trees. How many trees did the grove workers plant today?"
        Answer: "The answer is 6."


        Question2: {question}
        '''
        completion=palm.generate_text(model=model,prompt=prompt,temperature=0.5,max_output_tokens=1024)
        model_answer=completion.result
        # Write the question and the new answer to the text file
        file.write(f"QUESTION: {question}\n***********************************\nANSWER: {answer}\n**********************************\nMODEL's_ANSWER: {model_answer}\n##############################################################################################\n")
        
       
        
   


print("Text file 'evaluation_PaLM_1shot_standardPrompt.txt' created successfully.")

## Zero shot standsrd prompt

In [ ]:
import pandas as pd



# Open the text file in write mode
with open('evaluation_PaLM_0shot_standardPrompt.txt', 'w',encoding="utf-8") as file:
    # Iterate over each row in the DataFrame
    for index, row in df.iterrows():
        question = row['question']
        answer = row['answer']
        prompt=f'''
        Solve the following question:



        Question1: {question}
        '''
        completion=palm.generate_text(model=model,prompt=prompt,temperature=0.5,max_output_tokens=1024)
        model_answer=completion.result
        # Write the question and the new answer to the text file
        file.write(f"QUESTION: {question}\n***********************************\nANSWER: {answer}\n**********************************\nMODEL's_ANSWER: {model_answer}\n##############################################################################################\n")
        
       
        
   


print("Text file 'evaluation_PaLM_0shot_standardPrompt.txt' created successfully.")